src/script.ts

```diff
@@ -174,67 +245,18 @@ window.onload = function() {
         document.body.appendChild(tooltip);
     }
 
-    // Reload the page at 6am
-    const now = new Date();
-    let reloadTimer = new Date(now.getFullYear(), now.getMonth(), now.getDate(), 6, 0, 0, 0).getTime() - now.getTime();
-    if (reloadTimer < 0) {
-        reloadTimer += 24 * 60 * 60 * 1000;
-    }
-    setTimeout(location.reload, reloadTimer);
-
     // Update the clock every second
     setInterval(updateClock, 1000);
     updateClock();
 
     // Update the desktops every 2.5 seconds
-    setInterval(() => getDesktopsInUse().then(updateMap), 2500);
+    setInterval(() => getDesktopsInUse().then(updateDesktops), 2500);
 
     // Only update hours with the results from this request after the page has finished loading
     promiseGetHours.then((hoursArray) => {
-        const hours = hoursArray[0];
-        if (hours !== null) {
-            const start = document.getElementById("starthours");
-            const end = document.getElementById("endhours");
-
-            if (start == null) {
-                console.log("document has no start time");
-            } else {
-                start.textContent = shortTime(hours[0]);
-            }
-
-            if (end == null) {
-                console.log("document has no end time");
-            } else {
-                end.textContent = shortTime(hours[1]);
-            }
+        updateHours(hoursArray);
 
-            const closingTime = parseInt(hours[1].split(":")[0], 10);
-            const nightDate = new Date(now.getFullYear(), now.getMonth(), now.getDate(), closingTime, 0, 0, 0);
-            const nightTimer = nightDate.getTime() - now.getTime();
-            if (now.getHours() >= closingTime || now.getHours() <= 5) {
-                enableNightMode();
-            }
-            if (nightTimer > 0) {
-                setTimeout(enableNightMode, nightTimer);
-            }
-        } else {
-            enableNightMode();
-
-            const labhours = document.getElementById("labhours");
-
-            if (labhours == null) {
-                console.log("labhours element not found");
-            } else {
-                labhours.innerHTML = "";
-            }
-
-            const labtext = document.getElementById("labtext");
-
-            if (labtext == null) {
-                console.log("labtext element not found");
-            } else {
-                labtext.textContent = "Closed all day.";
-            }
-        }
+        // Every second, check if we need to switch to nightMode
+        setInterval(updateTheme, 1000);
```

Comment:
Perhaps instead of a bunch of different timers we can have one that fires every 2.5s (or 1s I guess, but every second seems like a lot of wasted cycles) and updates everything?

Reply:
I put this in a separate loop because I only want it to start after we get hours. Otherwise there will be a flicker effect where the map starts in day mode, then switches to night mode (since the hours haven't been loaded yet), then switches back to day mode (when the hours get loaded and we see that the lab is open)